File: sem06HW/binary_search.py
import random;

def binary_search(arr, target):
    start = 0
    end = len(arr) - 1
    found = False
    
    while start <= end:
        mid = (start + end) // 2
        if arr[mid] == target:
            found = True
            break
        elif arr[mid] < target:
            start = mid + 1
        else:
            end = mid - 1
    
    if found:
        return mid
    else:
        return None


def create_random_sort_list(length, min_value, max_value):
    random_list = []
    for _ in range(length):
        random_list.append(random.randint(min_value, max_value))
    random_list.sort()    
    return random_list


random_list = create_random_sort_list(10, 10, 20)
print(f'Список {random_list}')   
print(f'Индекс элемента 15 в списке: {binary_search(random_list, 15)}') 

File: sem06HW/BinarySearch.java
package sem06HW;

import java.util.Arrays;

public class BinarySearch {
    public static void main(String[] args) {
        int[] numbers = {1, 2, 5, 6, 8, 9};
        int target = 5;
        int result = binarySearch(numbers, target);
        
        System.out.println("Список: " + Arrays.toString(numbers));
        System.out.println("Ищем элемент: " + target);
        
        if (result != -1) {
            System.out.println("Элемент найден за индексом: " + result);
        } else {
            System.out.println("Элемент не найден в списке.");
        }
    }

    public static int binarySearch(int[] arr, int target) {
        int start = 0;
        int end = arr.length - 1;
        
        while (start <= end) {
            int mid = (start + end) / 2;
            if (arr[mid] == target) {
                return mid;
            } else if (arr[mid] < target) {
                start = mid + 1;
            } else {
                end = mid - 1;
            }
        }
        
        return -1;
    }
    
}